src/hooks/useClickOutside.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect } from 'react';
+
+export function useClickOutside(
+	isPartOfElement: (elem: Element | null) => boolean,
```

Comment:
Die isPartOfElement, volgens mij moet die logica in de hook zelf ingebakken zitten

Reply:
ik moet 2 elementen checken, de dropdown button en het flyout menu.
Dat zijn 2 aparte componenten. Een beetje als een portal. Dus je kan niet gewoon de component zelf checken door hem mee te geven aan de hook.

Als ik een wrapper toevoeg en die meegeef is die null aangezien de reference nog niet gecreert is at constructor time. En aangezien deze component een functional component is kan ik ook niet de componentDidMount lifecycle hook gebruiken.